### PR TITLE
[KYUUBI#5467] Integrate Intel Gluten with Spark engine

### LINKED ISSUE #5467: [TASK][MEDIUM] Integrate Intel Gluten with Spark engine
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before creating

- [X] I have searched in the [task list](https://github.com/orgs/apache/projects/296) and found no similar tasks.

### Mentor

- [X] I have sufficient knowledge and experience of this task, and I volunteer to be the mentor of this task to guide contributors to complete the task.

### Skill requirements

- Basic knowledge of Scala programming language, Kyuubi architecture, Apache Spark, Intel Gluten
- Basic English writing skills

### Background and Goals

[Gluten](https://github.com/oap-project/gluten) is a Spark plugin developed by Intel to accelerate Apache Spark with native libraries.

As Kyuubi currently supports using Spark as the computing engine, we can add tests to integrate with Gluten in the Spark engine, and write docs to guide users in setting up Kyuubi with Spark plus Gluten to improve performance.

### Implementation steps

- Add Gluten UTs in spark engine module, which could be conditionally enabled via maven profile and scalatest annotation
- Setup CI for Gluten testing
- Write docs to guide users in setting up Kyuubi with Spark plus Gluten.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Kwafoor**
# :mag: Description
## Issue References 🔗

This pull request fixes #5467 

## Describe Your Solution 🔧

1. Add Gluten UTs.
2. Setup CI for Gluten testing
3. Write docs to guide users in setting up Kyuubi with Spark plus Gluten.

## Types of changes :bookmark:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Related Unit Tests

github action ci tests: [Gluten Test CI](https://github.com/Kwafoor/incubator-kyuubi/actions/runs/7111586978)

---

# Checklists
## 📝 Author Self Checklist

- [ ] My code follows the [style guidelines](https://kyuubi.readthedocs.io/en/master/contributing/code/style.html) of this project
- [x] I have performed a self-review
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

## 📝 Committer Pre-Merge Checklist

- [ ] Pull request title is okay.
- [ ] No license issues.
- [ ] Milestone correctly set?
- [ ] Test coverage is ok
- [ ] Assignees are selected.
- [ ] Minimum number of approvals
- [ ] No changes are requested

**Be nice. Be informative.**

### CODE REVIEW

File: extensions/spark/kyuubi-spark-connector-tpcds/pom.xml [hunk 1]
```diff
@@ -213,4 +213,29 @@
         <outputDirectory>target/scala-${scala.binary.version}/classes</outputDirectory>
         <testOutputDirectory>target/scala-${scala.binary.version}/test-classes</testOutputDirectory>
     </build>
+
+    <profiles>
+        <profile>
+            <id>gluten</id>
```
Comment (ulysses-you) [inline extensions/spark/kyuubi-spark-connector-tpcds/pom.xml:219]: Can we make a new module for Gluten IT ? Includes tpcds and tpch related change.

**Review by ulysses-you — COMMENTED** [review]
thank you @Kwafoor for the work!

File: extensions/spark/kyuubi-spark-connector-tpcds/pom.xml (hunk 1 above)
Comment (Kwafoor) [reply extensions/spark/kyuubi-spark-connector-tpcds/pom.xml:219]: good idea, i will implement this.

**Comment by ulysses-you** [comment]
thanks, merged to master. We shoule keep check nightly build to see if the workflow work well.

**Comment by cfmcgrady** [comment]
late LGTM. thanks.

**Pull request closed without merging**